### ISSUE TITLE
Retry download

### ISSUE THREAD

**Issue by ermshiperete** [ermshiperete/BuildDependency#18]
Retry the download several times before failing if the server can't be reached. Currently it gets an exception and stops.

**Comment by rmunn** [comment]
A stack trace from one time when I saw this happen:

```text
Error executing task Dependencies: System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> System.ArgumentNullException: Value cannot be null.
Parameter name: source
  at System.Linq.Enumerable.FirstOrDefault[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate) [0x00010] in <63992662b765477a898ef49cdcc99ee2>:0 
  at BuildDependency.DependencyFile+<InternalLoadFile>d__10.MoveNext () [0x00253] in <5ee23ac1e05c4bbdae09735dde7188c9>:0 
   --- End of inner exception stack trace ---
  at System.Threading.Tasks.Task.ThrowIfExceptional (System.Boolean includeTaskCanceledExceptions) [0x00014] in <ffe5ec0b2c8a4c86831d8677cfb8b10e>:0 
  at System.Threading.Tasks.Task`1[TResult].GetResultCore (System.Boolean waitCompletionNotification) [0x00034] in <ffe5ec0b2c8a4c86831d8677cfb8b10e>:0 
  at System.Threading.Tasks.Task`1[TResult].get_Result () [0x0000b] in <ffe5ec0b2c8a4c86831d8677cfb8b10e>:0 
  at BuildDependency.DependencyFile.LoadFile (System.String fileName) [0x0000b] in <5ee23ac1e05c4bbdae09735dde7188c9>:0 
  at BuildDependency.Tasks.Dependencies+<ExecuteAsync>d__27.MoveNext () [0x0005d] in <5ee23ac1e05c4bbdae09735dde7188c9>:0 
   --- End of inner exception stack trace ---
  at System.Threading.Tasks.Task.ThrowIfExceptional (System.Boolean includeTaskCanceledExceptions) [0x00014] in <ffe5ec0b2c8a4c86831d8677cfb8b10e>:0 
  at System.Threading.Tasks.Task`1[TResult].GetResultCore (System.Boolean waitCompletionNotification) [0x00034] in <ffe5ec0b2c8a4c86831d8677cfb8b10e>:0 
  at System.Threading.Tasks.Task`1[TResult].get_Result () [0x0000b] in <ffe5ec0b2c8a4c86831d8677cfb8b10e>:0 
  at BuildDependency.Tasks.Dependencies.Execute () [0x001a2] in <5ee23ac1e05c4bbdae09735dde7188c9>:0 
  at Microsoft.Build.BuildEngine.TaskEngine.Execute () [0x00000] in <39da1c31b2094f29ab0836685e9d2b89>:0 
  at Microsoft.Build.BuildEngine.BuildTask.Execute () [0x0008f] in <39da1c31b2094f29ab0836685e9d2b89>:0 
---> (Inner Exception #0) System.AggregateException: One or more errors occurred. ---> System.ArgumentNullException: Value cannot be null.
Parameter name: source
  at System.Linq.Enumerable.FirstOrDefault[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate) [0x00010] in <63992662b765477a898ef49cdcc99ee2>:0 
  at BuildDependency.DependencyFile+<InternalLoadFile>d__10.MoveNext () [0x00253] in <5ee23ac1e05c4bbdae09735dde7188c9>:0 
   --- End of inner exception stack trace ---
  at System.Threading.Tasks.Task.ThrowIfExceptional (System.Boolean includeTaskCanceledExceptions) [0x00014] in <ffe5ec0b2c8a4c86831d8677cfb8b10e>:0 
  at System.Threading.Tasks.Task`1[TResult].GetResultCore (System.Boolean waitCompletionNotification) [0x00034] in <ffe5ec0b2c8a4c86831d8677cfb8b10e>:0 
  at System.Threading.Tasks.Task`1[TResult].get_Result () [0x0000b] in <ffe5ec0b2c8a4c86831d8677cfb8b10e>:0 
  at BuildDependency.DependencyFile.LoadFile (System.String fileName) [0x0000b] in <5ee23ac1e05c4bbdae09735dde7188c9>:0 
  at BuildDependency.Tasks.Dependencies+<ExecuteAsync>d__27.MoveNext () [0x0005d] in <5ee23ac1e05c4bbdae09735dde7188c9>:0 
---> (Inner Exception #0) System.ArgumentNullException: Value cannot be null.
Parameter name: source
  at System.Linq.Enumerable.FirstOrDefault[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate) [0x00010] in <63992662b765477a898ef49cdcc99ee2>:0 
  at BuildDependency.DependencyFile+<InternalLoadFile>d__10.MoveNext () [0x00253] in <5ee23ac1e05c4bbdae09735dde7188c9>:0 <---
<---
```

**Comment by ermshiperete** [comment]
We don't get an exception if the server can't be reached. The callstack is from a different scenario where the server can be reached but we can't find the specified project (cf #19). This has been fixed in versions >= 0.2.12).
